File: pulsar/java/src/main/java/com/examples/hello/pulsar/weather/conditions/Condition.java
package com.examples.hello.pulsar.weather.conditions;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Condition {
  // https://pulsar.apache.org/docs/en/client-libraries-java/#schema-example
  @Builder.Default
  Float temperature = 0.0f;
  @Builder.Default
  Float humidity = 0.0f;
}

File: pulsar/kotlin/src/main/kotlin/com/example/hello/pulsar/Condition.kt
package com.example.hello.pulsar

import kotlin.random.Random

data class Condition(var temperature: Float = 0f, var humidity: Float = 0f)
{
  fun randomize() {
    temperature = Random.nextFloat() * 40
    humidity = Random.nextFloat() * 100
  }
  companion object {
    fun random() = Condition(Random.nextFloat() *40, Random.nextFloat() * 100)
  }
}
